### PR TITLE
Fix install targets

### PR DESCRIPTION
**Pull request by oxidase**
# Issue

PR fixes incorrect installation paths and permissions.

Heads up: `libvalhalla.a` [packing is incorrect](https://github.com/valhalla/valhalla/blob/8aeb58a8bd3c9a244616a0f86f7b5f1395bdb235/src/CMakeLists.txt#L166-L177). 
Both static and shared libraries don't contain symbols of sub-modules.
This is a major issue and must be fixed before releasing the next version .

A possible way to fix the the issue is to use [`OBJECT` libraries](https://cmake.org/cmake/help/v3.9/manual/cmake-buildsystem.7.html#object-libraries)

/cc @gknisely 

## Tasklist

 - [ ] add tests
 - [ ] review - you must request approval to merge any PR to master
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Add to release notes in valhalla-docs
 - [ ] update relevant documentation if this is an API impacting change

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,10 +10,6 @@
 cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 project(valhalla LANGUAGES CXX C)
 
-if("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
```
Comment (TheMarex) [inline CMakeLists.txt]: Do you know why this was here? This seems like an odd case to explicitly check?

**Comment by kevinkreiser** [comment]
I'm having trouble finding where `make_locales.sh` gets called to generate the locale files needed for testing number formatting stuff that `odin` does for making instructions.

@oxidase can you point me to the right location that this should be happening? i had a look at `src/odin/CMakeLists.txt` and saw the transformation of the json. but we really need these locale files so that `std:locale*` can do some formatting. i realize that macos probably can't do this since it uses debian specific commands but its definitely something id like to figure out if you can give me a hand! maybe in the crossplatform future we'll be using a 3rd party lib for this.

File: CMakeLists.txt (hunk 1 above)
Comment (oxidase) [reply CMakeLists.txt]: I added recently this check in 20185625a9087e5dff5a4380b6c76e11772ee473 but it does not work as expected. The best would be to use `set(CMAKE_DISABLE_SOURCE_CHANGES ON)` but it is not possible with the current tests setup.

**Comment by oxidase** [comment]
@kevinkreiser locale files are generated in [`.circleci/config.yml`](https://github.com/valhalla/valhalla/blob/71c302d2c7fd4ecaee307848e31b4bafc4fcf1f8/.circleci/config.yml#L26) and it is not a part of CMake build. 

It is possible to add a target to generate locales, but it will be specific only for Ubunutu.

**Comment by oxidase** [comment]
as per chat with @gknisely after testing `make install` targets the pull request can be merged. The coverage drop will be investigated after merging.

**Pull request merged**